core/modules/hubotModule.js

```diff
@@ -0,0 +1,134 @@
+﻿/**
+ * Provides helper functions for handling user and system modules.
+ *
+ * Written By:
+ * 		Matthew Knox
+ *
+ * License:
+ *		MIT License. All code unless otherwise specified is
+ *		Copyright (c) Matthew Knox and Contributors 2015.
+ */
+
+var fs = require('fs'),
+    path = require('path'),
+    files = require.once('./../files.js'),
+    config = require('./../config.js'),
+    modulesDir = 'modules',
+    descriptor = 'hubot.json',
+    pkg = 'package.json',
+    Robot = require.once('./hubot/robot.js'),
+    coffeescriptLoaded = false;
+
+var verifyModuleDescriptior = function (hj, disabled) {
+    if (!hj.name || !hj.startup || !hj.version) {
+        return false;
+    }
+    
+    if (disabled === true && exports.disabledConfig 
+        && exports.disabledConfig[hj.name] && exports.disabledConfig[hj.name] === true) {
+        return false;
+    }
+    return true;
+};
+
+exports.verifyModule = function (location, disabled) {
+    var stat = fs.statSync(location);
+    if (!stat.isDirectory()) {
+        return null;
+    }
+    
+    var folderPath = path.resolve(location),
+        desc = path.join(folderPath, './' + descriptor),
+        pack = path.join(folderPath, './' + pkg),
+        hj;
+
+    try {
+        fs.statSync(desc);
+        hj = require.once(desc);
+    }
+    catch (e) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Closing curly brace does not appear on the same line as the subsequent block.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529645521/issues/source?bid=3282987&fileBranchId=3282997#l49)